### PR TITLE
added mean squared logarithic error

### PR DESCRIPTION
**Pull request by codeboy5**
Referenced from :- 
> https://scikit-learn.org/stable/modules/model_evaluation.html#mean-squared-log-error

@kartikdutt18 , hi could you have a look at the implementation, I can proceed to add the test after it.
Thanks!!!!

### CODE REVIEW

File: src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/**
+ * @file mean_squared_logarithmic_error.hpp
+ * @author Saksham Rastogi
+ *
+ * Definition of the mean squared logarithmic error performance function.
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp]: Could you remove performance from here.

File: src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  /**
+   * Computes the mean squared logarithmic error function.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param target The target vector.
+   */
+  template<typename InputType, typename TargetType>
+  double Forward(const InputType&& input, const TargetType&& target);
+  /**
```
Comment (kartikdutt18) [inline src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp]: Could you leave an extra line here.

**Comment by codeboy5** [comment]
@zoq , @kartikdutt18  the build shows 88 - RecurrentNetworkTest (Failed), any reason why this could happen since it compiles on other systems??(except Windows VS15 Plain)

**Comment by kartikdutt18** [comment]
Hi, don't worry about it, it's unrelated to your PR. It's on debug list of @zoq.

**Comment by kartikdutt18** [comment]
Thanks for the contribution!!!

File: HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ### mlpack ?.?.?
 ###### ????-??-??
+  * Added `mean squared logarithmic error` loss function (#2210).
```
Comment (rcurtin) [inline HISTORY.md]:
```suggestion
  * Added `mean squared logarithmic error` loss function for neural networks
    (#2210).
```

Just a quick fix; not everything in mlpack is neural networks, so it's nice to be specific. :)

File: src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * The mean squared logarithmic error performance function measures the network's
+ * performance according to the mean of squared logarithmic errors.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
```
Comment (say4n) [inline src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp]: Minor gripe, the `typename` should immediately follow the `<`, also no need for a new line at the end before the closing `>`!

File: src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp (hunk 2 above)
Comment (zoq) [reply src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp]: I unresolved the conversation, since the line didn't change.

**Comment by codeboy5** [comment]
@birm can you have a look and merge these changes ?

**Comment by zoq** [comment]
@codeboy5 can you fix the style issue, the the style check report?

**Comment by codeboy5** [comment]
> @codeboy5 can you fix the style issue, the the style check report?

Fixed the issue. Cheers 👍

**Pull request merged**